Signal/src/util/OWSBackupImportJob.m

```diff
@@ -371,97 +406,127 @@ - (void)restoreDatabaseContents:(OWSBackupStorage *)backupStorage completion:(OW
-    __block unsigned long long copiedAttachments = 0;
-
+    __block BOOL aborted = NO;
     [tempDBConnection readWriteWithBlock:^(YapDatabaseReadWriteTransaction *srcTransaction) {
         [primaryDBConnection readWriteWithBlock:^(YapDatabaseReadWriteTransaction *dstTransaction) {
-            // Copy threads.
-            [srcTransaction
-                enumerateKeysAndObjectsInCollection:[TSThread collection]
-                                         usingBlock:^(NSString *key, id object, BOOL *stop) {
-                                             if (self.isComplete) {
-                                                 *stop = YES;
-                                                 return;
-                                             }
-                                             if (![object isKindOfClass:[TSThread class]]) {
-                                                 OWSProdLogAndFail(
-                                                     @"%@ unexpected class: %@", self.logTag, [object class]);
-                                                 return;
-                                             }
-                                             TSThread *thread = object;
-                                             [thread saveWithTransaction:dstTransaction];
-                                             copiedThreads++;
-                                             copiedEntities++;
-                                         }];
-
-            // Copy attachments.
-            [srcTransaction
-                enumerateKeysAndObjectsInCollection:[TSAttachmentStream collection]
-                                         usingBlock:^(NSString *key, id object, BOOL *stop) {
-                                             if (self.isComplete) {
-                                                 *stop = YES;
-                                                 return;
-                                             }
-                                             if (![object isKindOfClass:[TSAttachment class]]) {
-                                                 OWSProdLogAndFail(
-                                                     @"%@ unexpected class: %@", self.logTag, [object class]);
-                                                 return;
-                                             }
-                                             TSAttachment *attachment = object;
-                                             [attachment saveWithTransaction:dstTransaction];
-                                             copiedAttachments++;
-                                             copiedEntities++;
-                                         }];
-
-            // Copy interactions.
+            if (![srcTransaction boolForKey:kOWSBackup_Snapshot_ValidKey
+                               inCollection:kOWSBackup_Snapshot_Collection
+                               defaultValue:NO]) {
+                DDLogError(@"%@ invalid database.", self.logTag);
```

Comment:
We should only be importing into an empty database, so warn if the database isn't empty...